src/wp-includes/rest-api/endpoints/class-wp-rest-widget-types-controller.php

```diff
@@ -397,6 +420,71 @@ public function get_item_schema() {
 		return $this->add_additional_fields_schema( $this->schema );
 	}
 
+	public function render( $request ) {
+		return array(
+			'preview' => $this->render_legacy_widget_preview_iframe(
```

Comment:
Would it make sense to return a response with `Content-type: text/html` rather than have all of the HTML inside this JSON string?

Reply:
I considered that for a moment and decided against it. The reason is that:
* In success case we get an HTML response with `Content-type: text/html` which means we need to set `parse: false` in the `apiFetch`
* But in the error case we would still get JSON, so `parse: false` means we need to parse manually and I don't like that

If you have some ideas how to make the JS part better, I'm all ears!

Reply:
Yeah let's stick with this format, it matches what we do in the block renderer endpoint.